# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

Для того чтобы сделать вывод от наличии зависимости расчитаем процент наличия долгов в зависимости от количества детей

In [22]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
(data.groupby('children')['debt'].mean() * 100)

children
0    7.543822
1    9.234609
2    9.454191
3    8.181818
4    9.756098
5    0.000000
Name: debt, dtype: float64

Можно заметить, что процент наличия долга у родителей несколько выше, чем у бездетных. Но родители с 5-ю детьми показывают чудеса планирования, стоит уточнить как много клиентов разных категорий у банка:

In [23]:
data.groupby('children')['debt'].count()

children
0    14091
1     4808
2     2052
3      330
4       41
5        9
Name: debt, dtype: int64

Кажется показатели родителей с 5-ю детьми не являются репрезентативными - их всего 9, в таком случае всего лишь при добавлении 1 клиента с долгами и 5-ю детьми показатель резко изменит значение до 10%. 

Рассчитаем разницу между вероятность появления долга у бездетных и средней вероятностью появления долга у родителей с детьми(категорию с 5-ю детьми игнорируем как не репрезентативную):

In [24]:
(data.groupby('children')['debt'].mean().loc[1:4].mean() - data.groupby('children')['debt'].mean().loc[0]) * 100

1.6128566423186288

**Вывод:** 

Из полученных значений можно сделать вывод о том, что наличие детей увеличивает вероятность возникновения долга в среднем на  1,6%. Количество детей влият на вероятность возникновения долга значительно меньше.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

Расчитаем процент наличия долгов в зависимости от семейного положения:

In [25]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
(data.groupby('family_status')['debt'].sum() / data.groupby('family_status')['debt'].count()) \
.sort_values(ascending=False) * 100

family_status
Не женат / не замужем    9.763948
гражданский брак         9.313014
женат / замужем          7.560558
в разводе                7.064760
вдовец / вдова           6.624606
Name: debt, dtype: float64

Статистика говорит о том, что в каждой категории есть клиенты имеющие долги.
Проверим репрезентативность выборки:

In [26]:
data.groupby('family_status')['debt'].count()

family_status
Не женат / не замужем     2796
в разводе                 1189
вдовец / вдова             951
гражданский брак          4134
женат / замужем          12261
Name: debt, dtype: int64

**Вывод:** 
Расчеты показывают, что между семейным положением и возвратом кредитов есть зависимость. Наиболее рисковые группы - "Не женат / не замужем" и "в разводе", а наименее - "женат / замужем". Разница в вероятности возникновения долга между наиболее и наименее рисковыми группами составляет 3,1%.

В зависимости от оценки рисков их можно сгруппировать по баллам и добавлять их(баллы) для расчета финальной категории риска возникновения долга.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

Рассчитаем процент наличия долгов в зависимости от уровня дохода.

In [27]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
(data.groupby('total_income_category')['debt'].sum() / data.groupby('total_income_category')['debt'].count()) \
.sort_values(ascending=False) * 100

total_income_category
E    9.090909
C    8.498210
A    8.000000
B    7.060231
D    6.017192
Name: debt, dtype: float64

Проверим репрезентативность выборки:

In [28]:
data.groupby('total_income_category')['debt'].count()

total_income_category
A       25
B     5014
C    15921
D      349
E       22
Name: debt, dtype: int64

**Вывод:** 
Категория граждан D является наиболее кредитоспособной, выборка по ним также вполне репрезентативна.
Следом идут категории B, A, C в порядке возрастания риска возникновения долга. Выборка по ним так же репрезентативна. Исключение составляет
лишь категория A, но скорее всего клиентов с таким уровнем дохода в целом меньше всех.
Категория клиентов E не может считаться репрезентативной, но в целом на нее стоит обратить особое внимание. Возможно такие клиенты берут меньше кредитов или банк чаще отказывает им из-за рисков невозврата, поэтому и клиентов этой категории выборке мало.

В зависимости от оценки рисков их можно сгруппировать по баллам и добавлять их(баллы) для расчета финальной категории риска возникновения долга.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

Рассчитаем процент наличия долгов в зависимости от цели кредита.

In [29]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
(data.groupby('purpose_category')['debt'].sum() / data.groupby('purpose_category')['debt'].count()) \
.sort_values(ascending=False) * 100

purpose_category
операции с автомобилем      9.347978
получение образования       9.252758
проведение свадьбы          7.911803
операции с недвижимостью    7.255139
Name: debt, dtype: float64

Проверим репрезентативность выборки:

In [30]:
data.groupby('purpose_category')['debt'].count()

purpose_category
операции с автомобилем       4279
операции с недвижимостью    10751
получение образования        3988
проведение свадьбы           2313
Name: debt, dtype: int64

**Вывод:** 
Выборка по всем целям кредита репрезентативна.
Наибольший риск в группах берущих кредит на транспорт и образование. Наименьший на проведение свадьбы и операций с недвижимостью.

В зависимости от оценки рисков их можно сгруппировать по баллам и добавлять их(баллы) для расчета финальной категории риска возникновения долга.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

При изначальном выводе информации о таблице можно было заметить, что данные отсутствуют в двух столбцах: `"days_employed"` и `"total_income"`. Причем количество пропущенных данных совпадает. 
Чтобы выявить закономерность выведем данные из пропущеных ячеек:

In [41]:
data_na = pd.read_csv('/datasets/data.csv')
data_na = data_na[data_na['days_employed'].isna()]
display(data_na.head())

children  days_employed  dob_years education  education_id  \
12         0            NaN         65   среднее             1   
26         0            NaN         41   среднее             1   
29         0            NaN         63   среднее             1   
41         0            NaN         50   среднее             1   
55         0            NaN         54   среднее             1   

            family_status  family_status_id gender  income_type  debt  \
12       гражданский брак                 1      M    пенсионер     0   
26        женат / замужем                 0      M  госслужащий     0   
29  Не женат / не замужем                 4      F    пенсионер     0   
41        женат / замужем                 0      F  госслужащий     0   
55       гражданский брак                 1      F    пенсионер     1   

    total_income                           purpose  
12           NaN                   сыграть свадьбу  
26           NaN                       образование  
29           NaN  строительство жилой недвижимости  
41           NaN  сделка с подержанным автомобилем  
55           NaN                   сыграть свадьбу

Проверим, все ли строки с пропущенными данными мы отобрали.

In [46]:
data_na.info()
#data_na['children'].unique()
#data_na['education'].unique()
#data_na['family_status'].unique()
#data_na['dob_years'].unique()


<class 'pandas.core.frame.DataFrame'>
Int64Index: 2174 entries, 12 to 21510
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          2174 non-null   int64  
 1   days_employed     0 non-null      float64
 2   dob_years         2174 non-null   int64  
 3   education         2174 non-null   object 
 4   education_id      2174 non-null   int64  
 5   family_status     2174 non-null   object 
 6   family_status_id  2174 non-null   int64  
 7   gender            2174 non-null   object 
 8   income_type       2174 non-null   object 
 9   debt              2174 non-null   int64  
 10  total_income      0 non-null      float64
 11  purpose           2174 non-null   object 
dtypes: float64(2), int64(5), object(5)
memory usage: 220.8+ KB


Количество строк совпадает с количеством пропусков, а столбцы `"days_employed"` и `"total_income"` содержат только пропуски. следовательно,` data_na` содержит все строки с пропусками.

*Ответ:* 

Среди пропущенных данных имеется лишь корреляция между столбцами `"days_employed"` и `"total_income"`. Учитывая, что исходная таблица содержит дробные значения в колонке дохода, что нехарактерно для записи сумм, можно предположить, что это значение рассчитывается на основании трудового стажа клиента. В таком случае, если трудовой стаж не указан, то и расчет дохода не может быть произведен.

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* 
По сравнению со средним значением применение медианного менее всего искажает данные для количественных переменных. Особенно это заметно, когда в ряде данных имеются выделяющиеся значения, при расчете среднего они могут значительно исказить статистику и повлиять на общий вывод. В случае же применения медианного значения такие выделяющиеся значения не скажутся на общей картине, а значит сделанный вывод будет точнее.

### Шаг 4: общий вывод.

#### Текст исходной задачи:

Заказчик — кредитный отдел банка. Нужно разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов.
Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

#### Общий вывод

Для проведения анализа были использованы данные по 21525 клиентам банка включающие возраст, пол, уровень образования и дохода, а также семейное положение и сведения о наличии детей. Предварительный анализ показал наличие 9,9% (2174) клиентов с пропущенными данными по трудовому стажу и доходу. Учитывая значительную долю отсутствующей информации допускается некоторое отклонение полученных результатов от реальной ситуации.

Изначально были поставлены задачи по выявлению зависимости погашения кредита от двух факторов:
+ количества детей;
+ семейного положения.

В ходе анализа также были проверены другие факторы:
+ уровень дохода;
+ цель кредита.

##### Полученные результаты:
**1) Количество детей клиента не является показательным для расчета риска возникновения долга, однако само наличие детей увеличивает риск в среднем на 1.6%. Категория клиентов с 5-ю детьми не является репрезентативной, так как таких клиентов у банка мало (9).**

| Количество детей  | Процент наличия долга |
| :-------------- | ------------- |
| Нет детей  | 7.54%  |
| 1 ребенок  | 9.23%  |
| 2 ребенка   | 9.45%  |
| 3 ребенка  | 8.18%  |
| 4 ребенка  | 9.76%  |

            
Можно предположить, что наличие детей делает более сложным планирование клиентом своего бюджета, из-за чего и увеличивается вероятность риска.

**2) Анализ показал наличие зависимости долга по кредиту от семейного положения, разница между категориями клиентов составляет до 3.1% вероятности возникновения долга.**

| Семейное положение      | Процент наличия долга |
| :--------------      | ------------- |
| Не женат / не замужем  | 9.76%  |
| гражданский брак       | 9.31%  |
| женат / замужем       | 7.56%  |
| в разводе             | 7.06%  |
| вдовец / вдова         | 6.62%  |
   
          
Полученные значения наводят на предположение, что клиенты с меньшим жизненным опытом склонны к просрочке кредитов. И наоборот, имеющие опыт преодоления жизненных проблем являются более отвественными заемщиками.

**3) Зависимость наличия долга от дохода присутствует, разница между категориями составляет до 3%.**

| Категория дохода | Сумма дохода          | Процент наличия долга |
| :------------ | :--------------     | ------------- |
| Е             | 0 - 30 000          | 9.09%  |
| С             | 50 001 - 200 000    | 8.49%  |
| А             | 1 000 001+          | 8.00%  |
| В             | 200 001 - 1 000 000 | 7.06%  |
| D             | 30 0001 - 50 000    | 6.01%  |


Категория граждан D является наиболее кредитоспособной. Следом идут категории B, A, C в порядке возрастания риска возникновения долга. Причем клиентов с категорией А мало (25), отчего есть вероятность большой погрещности в расчетном риске этой категории. Таким образом следует сделать вывод об отсутствии прямой зависимости уменьшения риска возникновения долга при увеличении суммы дохода клиентов. 

Категория клиентов E согласно анализу является наиболее рисковой, но не может считаться репрезентативной так как таких клиентов у банка очень мало (22). Возможно такие клиенты берут меньше кредитов или банк чаще отказывает им из-за рисков невозврата (расчетный процент также оказался наибольшим), поэтому и клиентов этой категории в выборке мало.

**4) Увеличение риска возникновения долга от целей кредита подтверждается. Разница между группами составляет до 2%.**

| Категория дохода          | Процент наличия долга |
| :------------          | ------------- |
| операции с автомобилем     | 9.35%  |
| получение образования       | 9.25%  |
| проведение свадьбы         | 7.91%  |
| операции с недвижимостью   | 7.26%  |

Наибольший риск в группах берущих кредит на транспорт и образование. Наименьший на проведение свадьбы или операции с недвижимостью.

##### Рекомендации

При построении модели скоринга можно присваивать баллы "риска" за определенное увеличение вероятности риска клиента согласно проведенному анализу. Например, пусть 1 балл соответствет увеличению риска на 0.1%, а таком случае клиент с детьми получает 16 баллов риска, не женатые/ не замужем 31 балл. 

Для оценки влияния нескольких факторов следует провести дополнительный анализ и определить их взаимное влияние.

Повторный анализ, при условии заполнения пропущенных данных, также может уточнить общую ситуацию и выявить ранее упущенные закономерности.